R/Lrnr_glm_semiparametric

```diff
@@ -0,0 +1,251 @@
+#' \code{\link[stats]{glm.fit}}. It supports models of the form `linkfun(E[Y|A,W]) = linkfun(E[Y|A=0,W]) + A * f(W)` where `A` is a binary or continuous interaction variable,
+#' and `f(W)` is a user-specified parametric function (e.g. `f(W) = model.matrix(formula_sp, W)`). The baseline function `E[Y|A=0,W]` is fit using a user-specified Learner (possibly pooled over values of `A` and then projected onto the semiparametric model).
+#'
+#' @docType class
+#'
+#' @importFrom R6 R6Class
+#'
+#' @export
+#'
+#' @keywords data
+#'
+#' @return A learner object inheriting from \code{\link{Lrnr_base}} with
+#'  methods for training and prediction. For a full list of learner
+#'  functionality, see the complete documentation of \code{\link{Lrnr_base}}.
+#'
+#' @format An \code{\link[R6]{R6Class}} object inheriting from
+#'  \code{\link{Lrnr_base}}.
+#'
+#' @family Learners
+#'
+#' @section Parameters:
+#' \describe{
+#'   \item{\code{formula_sp}}{ A \code{\link{formula}} object specifying the parametric component of the semiparametric model.
+#'   }
+#'   \item{\code{lrnr_baseline}}{A baseline learner for estimation of the nonparametric component. This can be pooled or unpooled by specifying \code{return_matrix_predictions}}
+#'   \item{\code{interaction_variable = "A"}}{A interaction variable name (that can be found in `training_task$data`) to multiply by the design matrix generated by \code{formula_sp}. If NULL then the interaction variable is treated identically `1`.
+#'   In many applications, this will be the name of a binary treatment variable (e.g. `A`).
+#'   }
+#'   \item{\code{family = NULL}}{A family object whose link function specifies the type of semiparametric model (e.g. partially-linear least-squares (\code{\link{gaussian}}), partially-linear logistic regression (\code{\link{binomial}}), partially-linear log-linear regression (\code{\link{poisson}})
+#'    }
+#'   \item{\code{append_interaction_matrix = TRUE}}{Whether \code{lrnr_baseline} should be fit on `cbind(task$X,A*V)` where `A` is the interaction variable and `V` is the design matrix obtained from \code{formula_sp}.
+#'   Note, if `append_interaction_matrix = TRUE`, the resulting estimator will be projected onto the semiparametric model using \code{glm.fit}.
+#'   If this is FALSE and \code{interaction_variable} is binary then the semiparametric model is learned by stratifying on \code{interaction_variable}.
+#'   Specifically, if FALSE, \code{lrnr_baseline} is used to estimate `E[Y|A=0,W]` by subsetting to only observations with `A = 0`.
+#'   In the binary case, setting `append_interaction_matrix = TRUE` allows one to pool the learning across treatment arms and allows additive models to perform well.
+#'   }
+#'   \item{\code{return_matrix_predictions = FALSE}}{Only used if \code{interaction_variable} is binary. Whether to return a matrix output with three columns being `E[Y|A=0,W], E[Y|A=1,W], E[Y|A,W]`.
+#'   }
+#'  \item{\code{...}}{Not used.
+#'  }
+#' }
```

Comment:
Switch these to use the new markdown syntax that we're using in the `devel` branch. Again, `Lrnr_glmnet` gives a fine example for how to do this. Basically, you can get ride of all the LaTeX macros, like `\describe{}`, `\item{}`, etc. Also see the comment above about following the 80-column rule.